### ISSUE TITLE
Allow for torchvision models to pass dropout rate in the constructors

### ISSUE THREAD

**Issue by kirk86** [pytorch/vision#1959]
## 🚀 Feature

The current definition of models in torchvision (e.g. vgg) does not allow the user to pass a dropout rate in the model's constructor. We can pass in arguments like batch_norm, pretrained, etc. but cannot pass in user defined dropout rates.

## Motivation
There's no simple way to modify current torchvision models without copy-pasting the whole definition and changing it.
I'm referring to the slightest possible modifications here. For instance, I just want custom dropout rate, or maybe leaky relu instead of relu.

## Pitch
Flexibility in torch vision models would allow anyone involved in experimenting with common models to avoid code duplication and hacking, hence this will increase flexibility, readability and reproducibility.
How many times we come across scenarios where a vgg model has been copied from torchvision and then modified simply by adding dropout rates and leaky relus instead.

e.g.
```
from torchvision.models import vgg16
model = vgg16(pretrained=False, batch_norm=True, activations=[relu, leaky_relu, relu, leaky_relu], dropout_rates=[0.3, 0.3, 0.5, 0.7])
 ```
This increase readability and avoids duplication

## Additional context
[Further discussion in pytorch forums](https://discuss.pytorch.org/t/is-there-a-mistake-in-the-implementation-of-vgg-network-in-torchvision-models/72640)

**Comment by fmassa** [comment]
Hi,

We encourage users to copy-paste and modify the code because the space of possible modifications / configurations is huge, and trying to expose every possible configuration would be hopeless and make the code much harder to understand.
For example, what is each element of `activations` in your case? is it per block? What if we want to change it inside a block?

That being said, it is possible to perform the modifications you pointed out without having to copy-paste the whole definition of the model.

Here is an example:

```python
model = vgg16()
for m in model.modules():
    if isinstance(m, nn.Dropout):
        m.p = 0.3
```
or manually modify the layers you want, for example
```python
def my_vgg():
    model = vgg16()
    for i, m in enumerate(model.features[:10]):
        if isinstance(m, nn.ReLU()):
            m[i] = nn.LeakyReLU()
    return model

model = my_vgg()
```
and so on.

> How many times we come across scenarios where a vgg model has been copied from torchvision and then modified simply by adding dropout rates and leaky relus instead.

Each case is kind of unique, and I think that keeping simplicity of implementation is better than trying to be exhaustive in here. That's why torchvision tries to provide *reference implementations*, so that users can modify them for their own need. The more complicated the implementation, the harder it is to understand it and modify it.

I'm closing the issue, but let us know if you think differently.

**Comment by kirk86** [comment]
Thanks for the reply. 
I agree with all the scenarios you've raised and I'm in favour of simplicity rather than complexity, but my argument was that it makes it difficult to understand as it is what's the sole purpose of `**kwargs` in the signature definition of the method?

1.) As I understand it `**kwargs` is unpacked in into `init_weights`?
2.) Currently, is only used to unpack `num_classes=10,23,100, etc.`?

**Comment by fmassa** [comment]
@kirk86 the `**kwargs` is there to make for a kind of unified API for the models. Its use depend on the models (for example, resnet doesn't have `init_weights` but have other arguments)
